File: CS/Commands/DocumentCommands.cs
using DXRibbonNotepad.Model;
namespace DXRibbonNotepad.Commands {
    public static class DocumentCommands {
        public static readonly DocumentCommand NewDocumentCommand;
        public static readonly DocumentCommand CloseDocumentCommand;

        static DocumentCommands() {
            NewDocumentCommand = new DocumentCommand(NewDocument);
            CloseDocumentCommand = new DocumentCommand(CloseDocument, CanCloseDocument);
        }
        static bool CanCloseDocument(Documents documents) {
            return documents.SelectedIndex != -1;
        }
        static void CloseDocument(Documents documents) {
            documents.RemoveAt(documents.SelectedIndex);
        }
        static void NewDocument(Documents documents) {
            documents.Add(new Document());
        }
    }
}

File: VB/Commands/DocumentCommands.vb
Imports DXRibbonNotepad.Model

Namespace DXRibbonNotepad.Commands

    Public Module DocumentCommands

        Public ReadOnly NewDocumentCommand As DocumentCommand

        Public ReadOnly CloseDocumentCommand As DocumentCommand

        Sub New()
            NewDocumentCommand = New DocumentCommand(AddressOf NewDocument)
            CloseDocumentCommand = New DocumentCommand(AddressOf CloseDocument, AddressOf CanCloseDocument)
        End Sub

        Private Function CanCloseDocument(ByVal documents As Documents) As Boolean
            Return documents.SelectedIndex <> -1
        End Function

        Private Sub CloseDocument(ByVal documents As Documents)
            documents.RemoveAt(documents.SelectedIndex)
        End Sub

        Private Sub NewDocument(ByVal documents As Documents)
            documents.Add(New Document())
        End Sub
    End Module
End Namespace
